aws_lambda_powertools/tracing/tracer.py

```diff
@@ -153,7 +155,7 @@ def __init__(
         self.__build_config(
             service=service, disabled=disabled, auto_patch=auto_patch, patch_modules=patch_modules, provider=provider
         )
-        self.provider = self._config["provider"]
+        self.provider: BaseProvider = self._config["provider"]
```

Comment:
could you remove this as it has an unintended side effect for escape hatches in X-Ray? e.g. auto-completing methods that we're not providing in the BaseProvider.

Reply:
Backing up as it's a Pycharm thing only, not VSCode. Best to keep it, and then later I'll add non-ABC methods for the sake of completion